### PR TITLE
Added an es6 method for _.sortBy and _.orderBy

### PR DESCRIPTION
**Pull request by HermannBjorgvin**
This implementation is more limited than both Underscore and Lodash but worked in my case for what I had to do which was just to simply sort a list of objects alphabetically by an object key.

I don't have any information about browser compatibility, I guess the arrow functions and let keyword are the only es6 parts of this.

Cheers, thanks for the useful repo.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -117,7 +117,8 @@ then Lodash/Underscore is the better option.*
 1. [_.reduce](#_reduce)
 1. [_.reduceRight](#_reduceright)
 1. [_.size](#_size)
-1. [_.some](#_some)
+1. [_.some](#_some))
```
Comment (cht8687) [inline README.md]: There is an extra ')' after some

Comment (cht8687) [reply README.md]: _.orderBy seems missing from the quick links.

File: README.md [hunk 2]
```diff
@@ -117,7 +117,8 @@ then Lodash/Underscore is the better option.*
...
+1. [_.sortBy](#_sortBy)
```
Comment (cht8687) [inline README.md]: (#_sortBy) need to be changed to (#_sortby) to make the quick link work.

File: README.md [hunk 3]
```diff
@@ -1126,6 +1127,35 @@ Tests whether any of the elements in the array pass the test implemented by the
...
 
+### _.sortBy
+
+Sorts an array of object based on an object key provided by a parameter (note this is more limited than Underscore/Lodash).
+
+  ```js
+  const fruits = [
+    {name:"banana", amount: 2},
+    {name:"apple", amount: 4},
+    {name:"pineapple", amount: 2}
```
Comment (cht8687) [inline README.md]: missing ',' in the code.

File: README.md [hunk 4]
```diff
@@ -1126,6 +1127,35 @@ Tests whether any of the elements in the array pass the test implemented by the
...
+  // Underscore
+  _.sortBy(fruits, 'name');
+  // => [{name:"apple", amount: 4}, {name:"banana", amount: 2}, {name:"mango", amount: 1}, {name:"pineapple", amount: 2}]
+
+  // Lodash
+  _.orderBy(fruits, ['name'],['asc']);
+  // => [{name:"apple", amount: 4}, {name:"banana", amount: 2}, {name:"mango", amount: 1}, {name:"pineapple", amount: 2}]
+
+  // Native
+  let sortBy = (key) => {
```
Comment (cht8687) [inline README.md]: maybe use `const` is better here since sortBy never changed.

**Comment by cht8687** [comment]
Hi @HermannBjorgvin , thanks for the contribution. Please check those comments which I made.

File: README.md (hunk 4 above)
Comment (HermannBjorgvin) [reply README.md]: Yes definitely makes more sense.

**Comment by HermannBjorgvin** [comment]
Alright so I fixed the linking issues you mentioned. I changed let sortBy to const. Fixed those couple typos. I also changed the header to "_.sortBy and _.orderBy" but kept the two seperate links so it's easier to find alphabetically.

File: README.md [hunk 5]
```diff
@@ -116,8 +116,10 @@ then Lodash/Underscore is the better option.*
 1. [_.range](#_range)
 1. [_.reduce](#_reduce)
 1. [_.reduceRight](#_reduceright)
+1. [_.orderBy](#_sortby-and-_orderby)
```
Comment (cht8687) [inline README.md]: A tiny last change - need move this _.orderBy to up, after _.minBy and before _.plunk ... ( as it's alphabetically listed)

Comment (HermannBjorgvin) [reply README.md]: Ah shoot, moved it to the right spot.

**Comment by HermannBjorgvin** [comment]
@cht8687 fixed that alphabetical ordering :+1:

**Comment by petermikitsh** [comment]
It might be worth noting that lodash (haven't checked for underscore) returns a **copy** of the array (sorted), whereas the native sort API **mutates** the original.

If you want to mimic the behavior of lodash, you'd want to do (for example):

```js
const sortedArr = [...arr].sort(sortFn);
```

I used this guide when transitioning off of lodash, and that caveat would have been helpful to know, since I ended up introducing a regression into my code.

**Comment by HermannBjorgvin** [comment]
Sorry about that @petermikitsh, still I'm honored that you're the first person I know of that I've introduced a regression into their code :)

**Pull request merged**